concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,503 @@
+
+-- ------------------------------------------------------------------
+-- Title: Oxford Acute Severity of Illness Score (OASIS)
+-- This query extracts the Oxford acute severity of illness score.
+-- This score is a measure of severity of illness for patients in the ICU.
+-- The score is calculated on the first day of each ICU patients' stay.
+-- OASIS score was originally created for MIMIC
+-- This script creates a pivoted table containing the OASIS score in eICU 
+-- ------------------------------------------------------------------
+
+-- Authors:
+-- Tristan Struja, MD, MSc, MPH (ORCID 0000-0003-0199-0184) and João Matos, MS (ORICD 0000-0002-0312-1647)
+
+-- Reference for OASIS:
+--    Johnson, Alistair EW, Andrew A. Kramer, and Gari D. Clifford.
+--    "A new severity of illness scale using a subset of acute physiology and chronic health evaluation data elements shows comparable predictive accuracy*."
+--    Critical care medicine 41, no. 7 (2013): 1711-1718.
+-- https://alistairewj.github.io/project/oasis/
+
+-- Variables used in OASIS (first 24h only):
+--  Heart rate, MAP, Temperature, Respiratory rate
+--  (sourced FROM `physionet-data.eicu_crd_derived.pivoted_vital`)
+--  GCS
+--  (sourced FROM `physionet-data.eicu_crd_derived.pivoted_vital` and `physionet-data.eicu_crd_derived.physicalexam`)
+--  Urine output 
+--  (sourced  FROM `physionet-data.eicu_crd_derived.pivoted_uo`)
+--  Pre-ICU in-hospital length of stay 
+--  (sourced FROM `physionet-data.eicu_crd.patient`)
+--  Age 
+--  (sourced FROM `physionet-data.eicu_crd.patient`)
+--  Elective surgery 
+--  (sourced FROM `physionet-data.eicu_crd.patient` and `physionet-data.eicu_crd.apachepredvar`)
+--  Ventilation status 
+--  (sourced FROM `physionet-data.eicu_crd_derived.ventilation_events`, `physionet-data.eicu_crd.apacheapsvar`, 
+--   `physionet-data.eicu_crd.apachepredvar`, and `physionet-data.eicu_crd.respiratorycare`)
+
+-- Regarding missing values:
+-- Elective stay: If there is no information on surgery in an elective stay, we assumed all cases to be -> "no elective surgery"
+-- There are a lot of missing values, especially for urine output. Hence, we have created 2 OASIS summary scores:
+-- 1) No imputation, values as is with missings. 2) Imputation in case of NULL values, with 0's (common approach for severity of illness scores)
+
+-- Note:
+--  The score is calculated for *all* ICU patients, with the assumption that the user will subselect appropriate patientunitstayid.
+
+DROP TABLE IF EXISTS `physionet-data.eicu_crd_derived.pivoted_OASIS` ;
```

Comment:
Remove the drop table / create table statements, and rename the query to pivoted-oasis.sql